jackson-core/src/main/java/io/micronaut/jackson/core/tree/JsonNodeTreeCodec.java

```diff
@@ -94,7 +94,12 @@ public JsonNode readTree(JsonParser p) throws IOException {
                 }
             case VALUE_NUMBER_FLOAT:
                 if (config.useBigDecimalForFloats()) {
-                    return JsonNode.createNumberNode(p.getDecimalValue());
+                    // NaN and Inf can't be represented by BigDecimal. Note: isNaN returns true for Inf, too.
+                    if (p.isNaN()) {
```

Comment:
`NaN`... are we javascript developers now? :rofl:

Reply:
NaNNaNNaNNaN... Batman!

Reply:
LOL